### DISCUSSION TITLE
TypeError and ReferenceError with import and Rollup

### DISCUSSION THREAD

**Discussion by BradleyHogle** [alpinejs/alpine#1597] [1. Help]
Hi all, I'm pretty new to Alpine, trying to figure out how to get it working with a build step using Gulp and Rollup.

I kept getting an error about Alpine being undefined until I imported in the following way in my js file:

```
import Alpine from '../node_modules/alpinejs/dist/module.esm';
window.Alpine = Alpine;
Alpine.start();
```

I'm keeping things simple with the example from the V3 documentation, also in my js file:
```
Alpine.data('dropdown', () => ({
    open: false,
    toggle() {
        this.open = ! this.open
    }
}))
```

In my index.html, I have:
```
<div x-data="dropdown">
      <button @click="toggle">Click me</button>
      <div x-show="open">Hello</div>
 </div>
```
 
 In the browser console, I get two errors:
 1) Uncaught TypeError: Object.defineProperty called on non-object
 2) Uncaught (in promise) ReferenceError: dropdown is not defined

What am I missing? If the answer is to use the CDN, I originally got similar errors (thus trying the build step). I can re-create and post those if that's more useful.

**Comment by danddanddand** [comment]
At a glance it looks okay but if you installed Alpine with npm, which it looks like you did, you should be able to just go "import Alpine from 'alpinejs';

Also, confirm your Alpine.data(...) is happening before the Alpine.start(). 

If you want to try this experimental feature you might get a better stack trace. https://github.com/alpinejs/alpine/pull/1607

**Reply by BradleyHogle** [reply]
Thanks for the answer! It was a combination of a couple things.

First, to my own shame, my Alpine.start( ) was indeed before my Alpine.data( ). Thanks for your patience on that one, I must have seen an example in the wrong part of the documentation.

Second, the import still doesn't work with just 'alpinejs', but does work the way I mentioned in my original post. Not sure why, but it's working this way so I'll leave it and periodically test it.

Third, once I resolved those, some other code of mine wasn't working. It was actually a Rollup issue, which I'm posting here in case someone runs into it in the future:

https://github.com/rollup/rollup/issues/1518#issuecomment-374727013

Appreciate the help!
